controllers/keda/scaledobject_controller.go

```diff
@@ -303,7 +304,7 @@ func (r *ScaledObjectReconciler) checkTargetResourceIsScalable(ctx context.Conte
 			logger.Error(errScale, "Target resource doesn't expose /scale subresource", "resource", gvkString, "name", scaledObject.Spec.ScaleTargetRef.Name)
 			return gvkr, errScale
 		}
-		isScalableCache[gr.String()] = true
+		isScalableCache.Store(gr.String(), true)
 	}
```

Comment:
I'm not sure if this is all going to happen concurrently? if so, wouldn't you still need a lock to protect the codepath between the load and the store?

Reply:
This is trying to check if a given resource type implements /scale, and caches the result. It might have a bit of inefficiency in calling `Get` multiple times on the same resource type. But it should be fine in my opinion as there is a small number of resource types in general.

Reply:
@arschles it is a `sync.Map` and the concurrent acces should be handled properly: https://golang.org/src/sync/map.go

The (1) usecase should be the suitable for our needs:
> // The Map type is optimized for two common use cases: (1) when the entry for a given 
> // key is only ever written once but read many times, as in caches that only grow,
>// or (2) when multiple goroutines read, write, and overwrite entries for disjoint
> // sets of keys. In these two cases, use of a Map may significantly reduce lock
> // contention compared to a Go map paired with a separate Mutex or RWMutex.

I was concerned about the efficiency as well, but agree with @ahmelsayed  that it should be ok for us. We can always split this cache in 2 - a static one (go Map) with prepopulated types (Deployment, Statefulset...) that will be checked first and a dynamic one (sync.Map) that will be used if the type is not found in the static one. But in this case it seems to be overkill. :)